app/controllers/api/lists_controller.rb

```diff
@@ -21,10 +21,13 @@ def remove
     end
 
     def save
-      ingredients = current_api_user.kitchen_list.ingredients
-      params[:list][:ingredients].each do |ingredient|
-        if !ingredient.key?('id')
-          Ingredient.create(name: ingredient[:name], kitchen_list_id: 1, category_id: 1)
+      # p "Entered the save route"
+      # p "--------------------------------------------------"
+      saved_ingredients = current_api_user.kitchen_list.ingredients
+      params[:ingredients].each do |passed_ingredient|
+        # p "#{passed_ingredient["name"]}:  #{saved_ingredients.any? {|h| h["name"] == passed_ingredient["name"]}}"
```

Comment:
@hueblerw,

What's the difference between line 25 and 27 here? Why are we expecting params to change? Did you change the front end?